app/forms/field_handlers/number_handler.py

```diff
@@ -65,8 +65,17 @@ def _field_type(
         )
 
     def get_field(self) -> Union[DecimalField, IntegerField]:
-        return self._field_type(
-            label=self.label, validators=self.validators, description=self.guidance
+        return (
+            self._field_type(
+                label=self.label,
+                validators=self.validators,
+                description=self.guidance,
+                places=self.max_decimals,
+            )
+            if self._field_type == DecimalFieldWithSeparator
+            else self._field_type(
+                label=self.label, validators=self.validators, description=self.guidance
+            )
         )
```

Comment:
I recommend only make the additional args conditional. Make it less verbose and fewer refactor needed if things change.

```py
        additional_args = (
            {"places": self.max_decimals}
            if self._field_type == DecimalFieldWithSeparator
            else {}
        )
        return self._field_type(
            label=self.label,
            validators=self.validators,
            description=self.guidance,
            **additional_args,
        )
```

Reply:
`get_field` refactored as suggested.